NSString+URLEncode.m

```diff
@@ -14,22 +14,24 @@ - (NSString *)URLEncode {
 }
 
 - (NSString *)URLEncodeUsingEncoding:(NSStringEncoding)encoding {
-    return (__bridge_transfer NSString *)CFURLCreateStringByAddingPercentEscapes(NULL,
-                                                                                 (__bridge CFStringRef)self,
-                                                                                 NULL,
-                                                                                 (CFStringRef)@"!*'\"();:@&=+$,/?%#[]% ",
-                                                                                 CFStringConvertNSStringEncodingToEncoding(encoding));
+    if ([self respondsToSelector:@selector(stringByAddingPercentEncodingWithAllowedCharacters:)]) {
+        NSCharacterSet *URLFullCharacterSet = [[NSCharacterSet characterSetWithCharactersInString:@" \"#%/:<>?@[\\]^`{|},()&!$'="] invertedSet];
+        return [self stringByAddingPercentEncodingWithAllowedCharacters:URLFullCharacterSet];
+    } else {
+        return [self stringByAddingPercentEscapesUsingEncoding:encoding];
```

Comment:
Right... I've just used it on iOS which always uses the if part and not the else